src/java.base/share/classes/java/io/BufferedInputStream.java

```diff
@@ -658,4 +662,22 @@ private long implTransferTo(OutputStream out) throws IOException {
         }
     }
 
+    /**
+     * Returns true if this class satisfies the following conditions:
+     * <ul>
+     * <li>does not retain a reference to the {@code byte[]}</li>
+     * <li>does not leak a reference to the {@code byte[]} to non-trusted classes</li>
+     * <li>does not modify the contents of the {@code byte[]}</li>
+     * <li>{@code OutputStream.write(byte[], int, int)} write does not read the contents outside of the offset/length bounds</li>
+     * </ul>
```

Comment:
The implementation change looks fine, just a typo at L671 where it says "write write", I think you can shorten this to say that the write method doesn't read the contents outside of the offset/length bounds.

Reply:
Done. Can we somehow modify the test to make it white-box one? Maybe it's possible to measure memory allocation before and after method invocation in the way that we could use the difference as a proof of non-allocating invocation for trusted OutputStreams?

Reply:
I assume you mean "to make it a blackbox"? Actually I do not see how we could do that *reliably*, as I already wrote recently.

Reply:
`com.sun.management.ThreadMXBean#getCurrentThreadAllocatedBytes` is reliable, isn't it?

Reply:
According to its JavaDocs it only returns *an approximation*, whatever that means. And it is measuring memory, while I already explained that the aim of this PR IMHO is not to spare memory, but to spare *time*. I wonder why we do not simply merge this PR as-is to get the actual benefits integrated ASAP instead of holding the train by heating up this discussion again?

Reply:
@mkarg apologies, of course I meant black box test.